File: benchmark/python/newton_sqrt.py
import time

def sqrt_newton(n):
    if n == 0:
        return 0
    x = n / 2.0
    for _ in range(20):
        x = (x + n / x) / 2.0
    return x

start = time.time() * 1000
s = 0.0
for i in range(1, 10001):
    s += sqrt_newton(i)
end = time.time() * 1000
print(f"Sum: {s}")
print(f"Time: {end - start:.0f}ms")

File: benchmark/lua/newton_sqrt.lua
local function sqrt_newton(n)
    if n == 0 then
        return 0
    end
    local x = n / 2.0
    for _ = 1, 20 do
        x = (x + n / x) / 2.0
    end
    return x
end

local start = os.clock()
local s = 0.0
for i = 1, 10000 do
    s = s + sqrt_newton(i)
end
local finish = os.clock()
print("Sum: " .. s)
print(string.format("Time: %.0fms", (finish - start) * 1000))

